src/GeneExpressionTSnePlot.js

```diff
@@ -93,13 +113,41 @@ const GeneExpressionScatterPlot = (props) => {
       }
     },
     chart: {
-      height: height
+      height: plotData.max==0 ?  height*0.95 : height,
     },
     title: {
       text: `Gene expression`
     },
-    legend: {
-      enabled: false
+    colorAxis: plotData.max==0 ? {} :
+    {
+      min: 0.1,
+      max: 10**dataScale-1,
+      type: 'logarithmic',
+      reversed: false,
+      //Dynamic stop where the last change colour is 100K
+      stop: colourSchema.map((val,idx) => {
+        return idx <= (Math.min(dataScale,colourSchemaLength) -1) ? [(idx+1)/Math.min(dataScale,colourSchemaLength),val] : []
+      }),
+      minColor:`rgb(215, 255, 255)`,
+      maxColor: dataScale>colourSchemaLength ? colourSchema[colourSchemaLength-1] : colourSchema[dataScale-1],
+      marker: {
+         color: '#c4463a'    
+      }
+    },
+    legend: plotData.max==0 ? {enabled: false} : 
+    {
+      title: {
+        text: "Expression level (TPM)"
+      },
+      floating: false,
+      align: "center",
+      symbolHeight: 5,
+      symbolWidth: 450
+    },
+    tooltip: {
+      positioner: function () {
+        return { x: 30, y: 50 };
+      },
     }
   }
 
```

Comment:
Remove the `chartClassName` and `gradient` variable assignments if no longer used.